pkg/rbd/controllerserver.go

```diff
@@ -368,100 +308,92 @@ func (cs *ControllerServer) ValidateVolumeCapabilities(ctx context.Context, req
 // in store
 // nolint: gocyclo
 func (cs *ControllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequest) (*csi.CreateSnapshotResponse, error) {
-
 	if err := cs.validateSnapshotReq(req); err != nil {
 		return nil, err
 	}
-	snapshotNameMutex.LockKey(req.GetName())
 
+	snapshotNameMutex.LockKey(req.GetName())
 	defer func() {
 		if err := snapshotNameMutex.UnlockKey(req.GetName()); err != nil {
 			klog.Warningf("failed to unlock mutex snapshot:%s %v", req.GetName(), err)
 		}
 	}()
 
-	// Need to check for already existing snapshot name, and if found
-	// check for the requested source volume id and already allocated source volume id
-	if exSnap, err := getRBDSnapshotByName(req.GetName()); err == nil {
-		if req.SourceVolumeId == exSnap.SourceVolumeID {
-			if err = storeSnapshotMetadata(exSnap, cs.MetadataStore); err != nil {
-				return nil, status.Error(codes.Internal, err.Error())
-			}
-
-			return &csi.CreateSnapshotResponse{
-				Snapshot: &csi.Snapshot{
-					SizeBytes:      exSnap.SizeBytes,
-					SnapshotId:     exSnap.SnapID,
-					SourceVolumeId: exSnap.SourceVolumeID,
-					CreationTime: &timestamp.Timestamp{
-						Seconds: exSnap.CreatedAt,
-					},
-					ReadyToUse: true,
-				},
-			}, nil
+	// Fetch source volume information
+	//var rbdVol *rbdVolume
+	rbdVol := new(rbdVolume)
```

Comment:
I think line 323 exist as part of memory allocation experiment :), please remove it.

Reply:
Removed.